docs/proposals/rollout/README.md

```diff
@@ -0,0 +1,308 @@
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+- Provide federated support for Rollout Strategies like Argo-Rollout, Flagger, and Kruise-Rollout.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story #1: Deployment Rolling Update with 2 maxUnavailable and 1 maxSurge
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: rollout-demo
+spec:
+  replicas: 10
+  strategy:
+    rollingUpdate:
+      maxUnavailable: 2
+      maxSurge: 1
+  template:
+    metadata:
+      labels:
+        app: rollout-demo
+    spec:
+      containers:
+        - name: main
+          image: nginx:v2 # upgrade rollout-demo from V1 -> V2
+  selector:
+    matchLabels:
+      app: rollout-demo
+---
+apiVersion: karmada.io/v1alpha2
+kind: ResourceBinding
+metadata:
+  name: rollout-demo-deployment
+spec:
+  clusters: # scheduled to 2 clusters
+    - name: cluster-1
+      replicas: 4
+    - name: cluster-2
+      replicas: 6
+```
+
+We expect that the Deployment Rolling Update with 2 maxUnavailable and 1 maxSurge will be scheduled to 2 clusters, and the rollout process will be consistent with the semantics in a single-cluster setting.
+During rolling processing, the number of unavailable Pods will not exceed 2, and the number of Pods will not exceed 11.
+
+#### Story #2: Batch Release with Partition
+```yaml
+apiVersion: apps.kruise.io/v1alpha1
+kind: CloneSet
+metadata:
+  name: rollout-demo
+spec:
+  replicas: 10
+  updateStrategy:
+    partition: 4
+    maxUnavailable: 2
+  template:
+    metadata:
+      labels:
+        app: rollout-demo
+    spec:
+      containers:
+        - name: main
+          image: nginx:v2 # upgrade rollout-demo from V1 -> V2
+  selector:
+    matchLabels:
+      app: rollout-demo
+---
+apiVersion: karmada.io/v1alpha2
+kind: ResourceBinding
+metadata:
+  name: rollout-demo-cloneset
+spec:
+  clusters: # scheduled to 2 clusters
+    - name: cluster-1
+      replicas: 3
+    - name: cluster-2
+      replicas: 7
+```
+
+
+We expect that the CloneSet cloud be paused when the number of Pods with the new image reaches 4, and the rollout process will be consistent with the semantics in a single-cluster setting. 
```

Comment:
```suggestion
We expect the CloneSet updating to be paused when the number of Pods with the new image reaches 6, and the rollout process should be consistent with the semantics in a single-cluster setting. 
```
I'm unfamiliar with the CloneSet, please correct me if I'm wrong.
Given [the partition](https://openkruise.io/docs/user-manuals/cloneset#partition) is the `desired number or percent of Pods in old revisions`, so I guess the CloneSet will be paused when the number of new Pods reaches 6(10 - partition). Isn't?

Reply:
> I'm unfamiliar with the CloneSet, please correct me if I'm wrong. Given [the partition](https://openkruise.io/docs/user-manuals/cloneset#partition) is the `desired number or percent of Pods in old revisions`, so I guess the CloneSet will be paused when the number of new Pods reaches 6(10 - partition). Isn't?

yes